File: data/mods/go/lib/robot.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

const (
	laby_name_Void    = "Void"
	laby_name_Wall    = "Wall"
	laby_name_Rock    = "Rock"
	laby_name_Web     = "Web"
	laby_name_Exit    = "Exit"
	laby_name_Unknown = "Unknown"
)

func input() string {
	reader := bufio.NewReader(os.Stdin)
	text, _ := reader.ReadString('\n')
	input := strings.TrimSpace(text)

	if input == "quit" {
		os.Exit(0)
	}

	return input
}

func laby_name_left() {
	fmt.Println("left")
	input()
}

func laby_name_right() {
	fmt.Println("right")
	input()
}

func laby_name_forward() {
	fmt.Println("forward")
	input()
}

func laby_name_take() {
	fmt.Println("take")
	input()
}

func laby_name_drop() {
	fmt.Println("drop")
	input()
}

func laby_name_escape() {
	fmt.Println("escape")
	input()
}

func laby_name_say(s string) {
	fmt.Println("say " + s)
	input()
}

func laby_name_look() string {
	fmt.Println("look")

	ans := input()

	switch ans {
	case "void":
		return laby_name_Void
	case "wall":
		return laby_name_Wall
	case "rock":
		return laby_name_Rock
	case "web":
		return laby_name_Web
	case "exit":
		return laby_name_Exit
	default:
		return laby_name_Unknown
	}
}

func init() {
	fmt.Println("start")
	input()
}

File: data/mods/python3/lib/robot.py
def look():
    "Expected responses from look(): void, wall, rock, web, exit."
    response = input("look\n")
    say(response)
    if response == "quit":
        exit(0)
    return response


def left():
    return input("left\n")


def right():
    return input("right\n")


def forward():
    return input("forward\n")


def take():
    return input("take\n")


def drop():
    return input("drop\n")


def escape():
    return input("escape\n")


def say(s):
    "Prompt string is returned and displays in the Messages panel."
    input("say " + s + "\n")


input("start\n")
